examples/async_tasks/external_source_external_thread.rs

```diff
@@ -0,0 +1,87 @@
+use bevy::prelude::*;
+use crossbeam_channel::{unbounded, Receiver};
+use rand::Rng;
+
+fn main() {
+    let (tx, rx) = unbounded::<u32>();
+    std::thread::spawn(move || loop {
+        // Everything here happens in another thread
+        // This is where you could connect to an external data source
+        tx.send(rand::thread_rng().gen_range(0..2000)).unwrap();
+        std::thread::sleep(std::time::Duration::from_millis(
+            rand::thread_rng().gen_range(0..200),
+        ));
+    });
+
+    App::new()
+        .add_event::<StreamEvent>()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(StreamReceiver(rx))
+        .add_startup_system(setup_system)
+        .add_system(read_stream_system)
+        .add_system(spawn_text_system)
+        .add_system(move_text_system)
+        .run();
+}
+
+struct StreamReceiver(Receiver<u32>);
+struct StreamEvent(u32);
+
+fn setup_system(mut commands: Commands) {
+    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
+}
+
+// This system reads from the receiver and sends events to Bevy
+fn read_stream_system(receiver: ResMut<StreamReceiver>, mut events: EventWriter<StreamEvent>) {
+    for from_stream in receiver.0.try_iter() {
+        events.send(StreamEvent(from_stream))
+    }
+}
+
+fn spawn_text_system(
+    mut commands: Commands,
+    mut reader: EventReader<StreamEvent>,
+    asset_server: Res<AssetServer>,
+    mut loaded_font: Local<Option<Handle<Font>>>,
+) {
+    let font = if let Some(font) = &*loaded_font {
+        font.clone()
+    } else {
+        let font = asset_server.load("fonts/FiraSans-Bold.ttf");
+        *loaded_font = Some(font.clone());
+        font
+    };
+    let text_style = TextStyle {
+        font,
+        font_size: 20.0,
+        color: Color::WHITE,
+    };
+    let text_alignment = TextAlignment {
+        vertical: VerticalAlign::Center,
+        horizontal: HorizontalAlign::Center,
+    };
+    for (per_frame, event) in reader.iter().enumerate() {
+        commands.spawn_bundle(Text2dBundle {
+            text: Text::with_section(format!("{}", event.0), text_style.clone(), text_alignment),
```

Comment:
Why not `event.0.clone()`?

Reply:
`event.0` is a `u32`

Reply:
:laughing: oops, fair enough

I guess `.to_string()` would be the call that's definitely always correct, although formatting it is probably fine.